massa-network-worker/src/messages.rs

```diff
@@ -98,277 +78,474 @@ pub enum Message {
-                res.extend(public_key.to_bytes());
-                res.extend(random_bytes);
-                VERSION_SERIALIZER.serialize(version, &mut res)?;
+                self.u32_serializer
+                    .serialize(&(MessageTypeId::HandshakeInitiation as u32), buffer)?;
+                buffer.extend(public_key.to_bytes());
+                buffer.extend(random_bytes);
+                self.version_serializer.serialize(version, buffer)?;
             }
             Message::HandshakeReply { signature } => {
-                res.extend(u32::from(MessageTypeId::HandshakeReply).to_varint_bytes());
-                res.extend(signature.to_bytes());
-            }
-            Message::Block(block) => {
-                res.extend(u32::from(MessageTypeId::Block).to_varint_bytes());
-                WRAPPED_SERIALIZER.serialize(block, &mut res)?;
+                self.u32_serializer
+                    .serialize(&(MessageTypeId::HandshakeReply as u32), buffer)?;
+                buffer.extend(signature.to_bytes());
             }
             Message::BlockHeader(header) => {
-                res.extend(u32::from(MessageTypeId::BlockHeader).to_varint_bytes());
-                WRAPPED_SERIALIZER.serialize(header, &mut res)?;
+                self.u32_serializer
+                    .serialize(&(MessageTypeId::BlockHeader as u32), buffer)?;
+                self.wrapped_serializer.serialize(header, buffer)?;
             }
             Message::AskForBlocks(list) => {
-                res.extend(u32::from(MessageTypeId::AskForBlocks).to_varint_bytes());
-                let list_len: u32 = list.len().try_into().map_err(|_| {
-                    ModelsError::SerializeError(
-                        "could not encode AskForBlocks list length as u32".into(),
-                    )
-                })?;
-                res.extend(list_len.to_varint_bytes());
-                for hash in list {
-                    res.extend(hash.to_bytes());
+                self.u32_serializer
+                    .serialize(&(MessageTypeId::AskForBlocks as u32), buffer)?;
+                self.u32_serializer
+                    .serialize(&(list.len() as u32), buffer)?;
+                for (hash, info) in list {
+                    buffer.extend(hash.to_bytes());
+                    let info_type = match info {
+                        AskForBlocksInfo::Info => BlockInfoType::Info,
+                        AskForBlocksInfo::Operations(_) => BlockInfoType::Operations,
+                    };
+                    self.u32_serializer
+                        .serialize(&u32::from(info_type), buffer)?;
+                    if let AskForBlocksInfo::Operations(ids) = info {
+                        let op_ids_serializer = OperationIdsSerializer::new();
+                        op_ids_serializer.serialize(&ids, buffer)?;
+                    }
+                }
+            }
+            Message::ReplyForBlocks(list) => {
```

Comment:
I have implemented it following the deserializer and it don't need the shared storage. Did I miss something ? @gterzian

Reply:
I'm wondering if we need this here(instead just having a `panic` as a placeholder and to make it clear it's not actually used), because of how the message is constructed in practice in `node_worker`(see `ToSend::ReplyForBlocksInfo`). 

Basically, in practice we don't actually serialize the `Message::ReplyForBlocks`, instead we "map" the `ToSend::ReplyForBlocksInfo` to the serialized form of the message. 

Actually maybe you can review whether that actually makes sense?

Reply:
I prefer having it in case we refactor but that personal decision I think